src/main/java/org/assertj/core/api/AbstractSoftAssertions.java

```diff
@@ -18,43 +18,35 @@
 import java.lang.reflect.Constructor;
 import java.util.List;
 
+import org.assertj.core.error.AssertionErrorCreator;
 import org.assertj.core.internal.Failures;
 
-public class AbstractSoftAssertions implements InstanceOfAssertFactories {
+public class AbstractSoftAssertions implements SoftAssertion, InstanceOfAssertFactories {
 
   protected final SoftProxies proxies;
 
   public AbstractSoftAssertions() {
     proxies = new SoftProxies();
   }
 
-  public <T, V> V proxy(Class<V> assertClass, Class<T> actualClass, T actual) {
+  private AssertionErrorCreator assertionErrorCreator = new AssertionErrorCreator();
+
+  @Override
+  public void assertAll() {
+    List<Throwable> errors = errorsCollected();
+    if (!errors.isEmpty()) throw assertionErrorCreator.multipleSoftAssertionsError(errors);
+  }
+  
+  @Override
+  public <V extends Assert<? extends V, ? extends T>, T> V proxy(Class<V> assertClass, Class<T> actualClass, T actual) {
```

Comment:
use `ASSERT` instead of `V` and `ACTUAL` instead of `T` for readability

Reply:
Done.